chain/consensus/hierarchical/actors/sca/sca_actor.go

```diff
@@ -634,3 +639,108 @@ func requireSuccessWithNoop(rt runtime.Runtime, msg types.Message, code exitcode
 	}
 	return false
 }
+
+func (a SubnetCoordActor) InitAtomicExec(rt runtime.Runtime, params *AtomicExecParams) *atomic.LockedOutput {
+	rt.ValidateImmediateCallerType(builtin.AccountActorCodeID)
+	var (
+		st  SCAState
+		c   cid.Cid
+		err error
+	)
+	rt.StateTransaction(&st, func() {
+		c, err = params.Cid()
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "error computing cid for atomic exec params")
+		execMap, err := adt.AsMap(adt.AsStore(rt), st.AtomicExecRegistry, builtin.DefaultHamtBitwidth)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "error getting exec map")
+		// Check if execution already exists.
+		_, found, err := getAtomicExec(execMap, c)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "error getting exec map")
+		if found {
+			rt.Abortf(exitcode.ErrIllegalArgument, "execution with cid %s already initialized", c)
+		}
+
+		// sanity-check: verify that all messages have same method and are directed to the same actor.
+		method := params.Msgs[0].Method
```

Comment:
Params is not checked before accessing its fields

Reply:
Good call. We should check here if `Msgs` or `LockedStates` are empty. Thanks!